src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -889,6 +889,10 @@ private void WorkspaceAdded(WorkspaceModel item)
         private void WorkspaceRemoved(WorkspaceModel item)
         {
             workspaces.Remove(workspaces.First(x => x.Model == item));
+
+            // If a workspace is removed, reset this property to false 
+            // so that Run Automatic checkbox is unchecked by default
+            DynamicRunEnabled = false;
```

Comment:
Let's not do this because by right the value that you set `DynamicRunEnabled` to, should be the value of `HomeSpace.DynamicRunEnabled`. If the `HomeSpace` that we are switching to has `DynamicRunEnabled` set to `true`, then this logic will reset that. I believe what we need here is `RaisePropertyChanged("DynamicRunEnabled")`.